src/components/PageLayout/LoginForm/LoginForm.tsx

```diff
@@ -1,64 +1,68 @@
+import {FC} from 'react'
+import {SubmitHandler, useForm} from 'react-hook-form'
 
-import {Login} from '@/types/api/generated/user'
+import {FormInput} from '@/components/FormItems/FormInput/FormInput'
 import {AuthContainer} from '@/utils/AuthContainer'
 
 import styles from './LoginForm.module.scss'
 
+type LoginFormValues = {
+  email: string
+  password: string
+}
+
+const defaultValues: LoginFormValues = {
+  email: '',
+  password: '',
+}
+
 interface ILoginForm {
   closeOverlay: () => void
 }
 
 export const LoginForm: FC<ILoginForm> = ({closeOverlay}) => {
-  const [formData, setFormData] = useState<Login>({email: '', password: ''})
-  const emailRef = useRef<HTMLInputElement>(null)
   const {login} = AuthContainer.useContainer()
+  const {
+    handleSubmit,
+    control,
+    formState: {errors},
+  } = useForm<LoginFormValues>({defaultValues})
 
-  useEffect(() => {
-    emailRef.current?.focus()
-  }, [])
-
-  const handleChange = (e: FormEvent<HTMLInputElement>) => {
-    const {name, value} = e.currentTarget
-    setFormData((prevData) => ({...prevData, [name]: value}))
-  }
-
-  const handleLogin: FormEventHandler<HTMLFormElement> = async (e) => {
+  const onSubmit: SubmitHandler<LoginFormValues> = async (data) => {
     // ak to tu nie je, robi to weird veci- prida to email a password do URL
     // ako query parametre, refreshne stranku a vobec nestihne pustit nas kod
-    e.preventDefault()
-    await login(formData, closeOverlay)
+    await login(data, closeOverlay)
   }
 
+  const requiredRule = {required: '* Toto pole nemôže byť prázdne.'}
+
   return (
-    <>
-      <form onSubmit={handleLogin}>
-        <div className={styles.loginForm}>
-          <div>
-            <input
-              ref={emailRef}
-              type="text"
-              name="email"
-              placeholder="e-mail"
-              value={formData.email}
-              onChange={handleChange}
-            />
-          </div>
-          <div>
-            <input
-              type="password"
-              name="password"
-              placeholder="heslo"
-              value={formData.password}
-              onChange={handleChange}
-            />
-          </div>
+    <form className={styles.loginForm}>
+      <FormInput
+        control={control}
+        name="email"
+        label="Email"
+        rules={{
+          ...requiredRule,
+          pattern: {
+            value: /^[\w%+.-]+@[\d.a-z-]+\.[a-z]{2,}$/iu,
+            message: '* Vložte správnu emailovú adresu.',
+          },
+        }}
+        fieldError={errors.email}
+      />
+      <FormInput
+        control={control}
+        name="password"
+        label="Heslo"
+        type="password"
+        rules={requiredRule}
+        fieldError={errors.password}
+      />
 
-          <button>
-            <span className={styles.underline}>Prihlásiť</span>
-          </button>
-        </div>
-      </form>
-    </>
+      <button onClick={handleSubmit(onSubmit)}>
```

Comment:
pozeram ze vlastne - spravne a asi podla html standardu by miesto struktury:
```js
<form ...>
  ...
  <button onClick={handleSubmit(onSubmit)}>...</button>
</form>
```
mala byt struktura:
```js
<form onSubmit={handleSubmit(onSubmit)} ...>
  ...
  <button type="submit">...</button>
</form>
```

Reply:
predpokladam, ze registracia na tom bude podobne

Reply:
hm.. neviem ani ja. povedal by som, ze skor by to mal malo byt, ako nie. nema clovek preco to robit a ked stlaci enter, mozno to fakt chce odoslat, mozno je niekde ku koncu, alebo nieco akurat opravil podla chyby... aj tak sa ti pusti len validacia, ked este nemas vyplneny form